### DISCUSSION TITLE
Get-ServiceNowRecord -Table 'cmdb_ci_computer' examples

### DISCUSSION THREAD

**Discussion by jwarren126** [Snow-Shell/servicenow-powershell#176] [Q&A] [answered]
Are there any good examples on retrieving filtered records from the computer table?  I'd like to automate reading getting data from this table.

**Comment by gdbarron** [comment]
Hi @jwarren126.  Did the examples for Get-ServiceNowRecord not get it done?  What are you looking to do?

**Reply by jwarren126** [reply]
Specifically I'm looking to retrieve a set of computer records from the cmdb_ci_computer table, but I'm struggling with the syntax of the -Filter parameter when trying to combine multiple filters.  It's not entirely clear how the -group option works in combination with -and and -or sets.

**Reply by gdbarron** [reply]
Instead of just and/or, sometimes you want to find items with multiple filters as well as other items with multiple filters at the same time.  Let's say, taking random properties of cmdb_ci_computer, you want `(can_print is true and monitor is false) or (cost greater than 500 and OS contains the word mac)`.  This would return results where `can_print is true and monitor is false` is true **OR** `cost greater than 500 and OS starts with the word Mac` is true.  It's as if you ran 2 queries and combined the results.  For this example, the filter would look like `-Filter @('can_print, '-eq', $true), 'and', @('monitor', '-eq', $false), 'group', @('cost', '-gt', '500'), 'and', @('os', '-startswith', 'Mac')`.

Here's the SN definition for `group`
![image](https://user-images.githubusercontent.com/11862024/150348755-ae7cd03d-ee23-4e2a-8321-06ef547a3d09.png)

Here's the example for `Get-ServiceNowRecord` which explains it.  If this can be made better, please submit an enhancement request or PR.
![Screen Shot 2022-01-20 at 9 03 34 AM](https://user-images.githubusercontent.com/11862024/150353598-f1daee7e-8493-4d09-954e-b6869be828b0.png)

**Reply by jwarren126** [reply]
> It's as if you ran 2 queries and combined the results.

Perfect explanation that helped make it clear.  Thank you so much!

**Comment by gdbarron** [comment, marked as answer]
answer above...
